Engine/DataFeeds/SubscriptionCollection.cs

```diff
@@ -117,17 +126,70 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
 
+        /// <summary>
+        /// Helper method to validate a configuration to be included in the fill forward calculation
+        /// </summary>
+        private static bool ValidateFillForwardResolution(SubscriptionDataConfig configuration)
+        {
+            return !configuration.IsInternalFeed && configuration.Resolution != Resolution.Tick;
+        }
+        /// <summary>
+        /// Gets and updates the fill forward resolution by checking specified subscription configurations and
+        /// selecting the smallest resoluton not equal to tick
+        /// </summary>
+        private void UpdateFillForwardResolution(bool adding, SubscriptionDataConfig configuration)
+        {
+            // Due to performance implications let's be jealous in updating the _fillForwardResolution
+            if (ValidateFillForwardResolution(configuration) &&
+                (
+                    (adding  // we are adding
+                    && configuration.Resolution.ToTimeSpan() != _fillForwardResolution.Value) // check if the new timeSpan is different
+                ||
+                    (!adding // We are removing
+                    && configuration.Resolution.ToTimeSpan() == _fillForwardResolution.Value // True: We are removing the resolution we were using
+                    && _subscriptions.All(x => x.Key.Resolution != configuration.Resolution))) // False: there is at least another one equal, no need to update
+                )
+            {
+                var configurations = adding ? _subscriptions.Keys.Concat(new[] { configuration }) : _subscriptions.Keys;
+                _fillForwardResolution.Value = configurations.Where(ValidateFillForwardResolution)
+                                                             .Select(x => x.Resolution)
+                                                             .Distinct()
```

Comment:
I know this is existing code that has been transplanted, but looking at it now I don't think we actually need this `Distinct` call.

Reply:
I added the `Distinct` call :P. Your right, we don't need it, but I believed it would be more performant since we would remove duplicates. What do you think?

Reply:
`Distinct` under the cover creates a new set, loops through all entries pushing them into the set, and ones that are successfully added move on to the next iterator -- my hunch is that it's extra work we don't need to do -- I doubt the savings will be statistically relevant since we've already minimized how often this method is called -- would probably only be noticeable if we continuously added/removed subscriptions with different resolutions requiring the fill forward resolution to be constantly updated.